sources/engine/Stride.Assets/Materials/MaterialAssetCompiler.cs

```diff
@@ -23,7 +23,7 @@ internal class MaterialAssetCompiler : AssetCompilerBase
         public override IEnumerable<BuildDependencyInfo> GetInputTypes(AssetItem assetItem)
         {
             yield return new BuildDependencyInfo(typeof(TextureAsset), typeof(AssetCompilationContext), BuildDependencyType.Runtime);
-            yield return new BuildDependencyInfo(typeof(MaterialAsset), typeof(AssetCompilationContext), BuildDependencyType.CompileAsset);
+            yield return new BuildDependencyInfo(typeof(MaterialAsset), typeof(AssetCompilationContext), BuildDependencyType.Runtime);
```

Comment:
I'm trying to understand the effect of this change to determine if it's fine to merge this PR and resolve the issue referenced.

I managed to find the following deeper explanation:

- `BuildDependencyType.Runtime` - the referenced asset is not needed to compile this asset, but it's needed at runtime to use the compiled content (eg. Models need Materials, who need Textures. But you can compile Models, Materials and Textures independently).
- `BuildDependencyType.CompileContent` - the referenced asset needs to be compiled before this asset, and the compiler of this asset needs to load the corresponding content generated from the referenced asset (eg. A prefab model, which aggregates multiple models together, needs the compiled version of each model it's referencing to be able to merge them).
- `BuildDependencyType.CompileAsset` - the referenced asset is read when compiling this asset because it depends on some of its parameter, but the referenced asset itself doesn't need to be compiled first (eg. Navigation Meshes need to read the scene asset they are related to in order to gather static colliders it contains, but they don't need to compile the scene itself).

From this I'm understanding that currently when you build a material asset it may look into other referenced material assets and read some of their parameters. Switching to Runtime mode will potentially change it in a way where those parameters aren't always available (not sure, tbh).

Something I've seen done is a combination `BuildDependencyType.Runtime | BuildDependencyType.CompileAsset` - if the issue would still be resolved with that I think it would be preferred with a comment added to investigate it further and referencing the issue and PR for context.

Reply:
Der Fehler ist auch mit der Kombination `BuildDependencyType.Runtime | BuildDependencyType.CompileAsset` behoben.